### PR TITLE
feat(listing): add new variant - INNO-1096

### PR DESCRIPTION
**Pull request by raduchiriac**
# PR description

https://ec-europa.github.io/ec-europa-theme/section-listing.html#kssref-listing-7

### List-item

[feat]: Add new variant and class `.ecl-list-item--navigation`
[feat]: Add new demo configuration
```diff
+const contextNavigation = require('@ecl/generic-component-list-item/data/demo--navigation');
```
## QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

* [x] `package.json` is up-to-date and `@ecl/[system]-base` is part of the dependencies
* [x] I have checked the dependencies
* [x] I have given the fractal status “ready” to my component
* [x] I have declared `@define mycomponent` in the SCSS file
* [x] I have specified `margin: 0;` on the CSS component
* [x] I have provided tests
* [x] I follow the naming guidelines
* [x] the component supports composition
* [x] there are no hardcoded strings (all content come from the context)
* [x] I have filled the README.md file (at least a few lines)
* [x] My component is listed in the root README
* [x] My PR has the right label(s)

### CODE REVIEW

**Comment by HAL-Patch-INNO** [comment]
Deploy preview for *europa-component-library* ready!

Built with commit 31568c7825a6a630903d44bc1ceec40806344a2b

https://deploy-preview-707--europa-component-library.netlify.com

File: src/generic/generic-component-list-item/generic-component-list-item.twig [hunk 1]
```diff
@@ -70,11 +70,13 @@
       </div>
     {% endif %}
     <div class="ecl-list-item__body">
-      <div class="ecl-list-item__meta">
-        {% block meta_block %}
-          {% include '@ecl/generic-component-meta' with { metas: _metas } only %}
-        {% endblock %}
-      </div>
+      {% if _variant != 'navigation' %}
```
Comment (degliwe) [inline src/generic/generic-component-list-item/generic-component-list-item.twig]: Why not ` {% if _variant == 'thumbnail' %} ` ?

**Review by degliwe — CHANGES_REQUESTED** [review]
To be added:
- update tpl doc: https://github.com/ec-europa/europa-component-library/pull/707/files#diff-7f6b04e77b01ba90a3d4302bd2de8104L3
- update documentation ec/components/detail/ec-component-listing (when (not) to use this component) as per https://webgate.ec.europa.eu/CITnet/jira/browse/INNO-1096?focusedCommentId=2741480&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2741480
- add navigation list item to ec/components/detail/ec-template-listing
- add navigation list item example to ec-component-listing#playground

File: src/generic/generic-component-list-item/generic-component-list-item.twig (hunk 1 above)
Comment (raduchiriac) [reply src/generic/generic-component-list-item/generic-component-list-item.twig]: but I see the `default` variant also uses metas, not sure about the rest but navigation will never need meta so I used "all but navigation" test

Comment (raduchiriac) [reply src/generic/generic-component-list-item/generic-component-list-item.twig]: what about the `default` variant?

File: src/generic/generic-component-list-item/generic-component-list-item.scss [hunk 2]
```diff
@@ -94,4 +94,40 @@
...
     padding-right: map-get($ecl-spacing, 's');
   }
+
+  // Navigation item
+  .ecl-list-item--navigation {
+    margin-bottom: map-get($ecl-spacing, 's');
+  }
+
+  .ecl-list-item--navigation .ecl-list-item__link {
+    border-top: 3px solid map-get($ecl-colors, 'yellow-100');
```
Comment (degliwe) [inline src/generic/generic-component-list-item/generic-component-list-item.scss]: why 3px?

File: src/generic/generic-component-list-item/generic-component-list-item.twig [hunk 3]
```diff
@@ -70,11 +70,13 @@
       </div>
     {% endif %}
     <div class="ecl-list-item__body">
-      <div class="ecl-list-item__meta">
-        {% block meta_block %}
-          {% include '@ecl/generic-component-meta' with { metas: _metas } only %}
-        {% endblock %}
-      </div>
+      {% if _variant not in ['navigation'] %}
```
Comment (degliwe) [inline src/generic/generic-component-list-item/generic-component-list-item.twig]: 'default' is gone? if you compare only one value, why not `{% if _variant  == 'navigation' %}` ?

**Review by degliwe — CHANGES_REQUESTED** [review]

**Review by degliwe — APPROVED** [review]

**Pull request merged**